api/interface.py

```diff
@@ -72,6 +72,20 @@ class AudioSettings(BaseModel):
 class SettingsConfig(BaseModel):
     audio: Optional[AudioSettings] = None
 
+class WhispercppAutostartSettingsConfig(BaseModel):
+    whispercpp_exe_path: Optional[str] = None
+    whispercpp_model_path: Optional[str] = None
+
+class WhispercppSttConfig(BaseModel):
+    base_url: str
+    autostart: bool
+    autostart_settings: WhispercppAutostartSettingsConfig
```

Comment:
I think in this case `autostart_settings` should be optional (so that we could omit it in the config but if we set it, `exe_path` and `model_path` are required. Correct?
You actually can't define a required object without defining all-optional props in yaml. It would be something like

```yaml
autostart_settings:
# nothing here because all optional (?)
```

But that can't be parsed and is invalid markup.